src/robusta/core/playbooks/internal/ai_integration.py

```diff
@@ -0,0 +1,70 @@
+import json
+import logging
+
+import requests
+
+from robusta.core.model.base_params import AIInvestigateParams
+from robusta.core.model.events import ExecutionBaseEvent
+from robusta.core.playbooks.actions_registry import action
+from robusta.core.reporting import Finding, FindingSubject
+from robusta.core.reporting.base import EnrichmentType
+from robusta.core.reporting.consts import FindingSubjectType, FindingType
+from robusta.core.reporting.holmes import HolmesRequest, HolmesResult, HolmesResultsBlock
+from robusta.integrations.prometheus.utils import HolmesDiscovery
+
+
+def build_investigation_title(investigation__type: str, params: AIInvestigateParams) -> str:
+    if investigation__type == "analyze_problems":
+        return params.ask
+
+    return params.context.get("issue_type", "unknown health issue")
+
+
+@action
+def ask_holmes(event: ExecutionBaseEvent, params: AIInvestigateParams):
+    holmes_url = HolmesDiscovery.find_holmes_url(params.holmes_url)
+    if not holmes_url:
+        logging.error("Holmes url not found")
+        return
+
+    investigation__title = build_investigation_title(params.investigation_type, params)
+    try:
+        holmes_req = HolmesRequest(
+            source=params.context.get("source", "unknown source"),
+            title=f"{investigation__title}",
+            subject=params.resource.dict() if params.resource else None,
```

Comment:
i think subject might not be able to be none via pydantic so this line might need to be
`subject=params.resource.dict() if params.resource else {}`